### PR TITLE
Adding Shuffle().

### PR DESCRIPTION
**Pull request by DeepakRajendrakumaran**
This PR adds `Vector512.Shuffle()`. 

1. `Vector256.Shuffle()` has a bug when index is out of range. I didn't change it for now. I worked around it for `Vector512`
2. `Vector512.IsHardwareAccelerated` already returns 1

@dotnet/avx512-contrib

### CODE REVIEW

**Comment by BruceForstall** [comment]
> 1. `Vector256.Shuffle()` has a bug when index is out of range. I didn't change it for now. I worked around it for `Vector512`

Can you open an issue for this?

**Comment by DeepakRajendrakumaran** [comment]
> > 1. `Vector256.Shuffle()` has a bug when index is out of range. I didn't change it for now. I worked around it for `Vector512`
> 
> Can you open an issue for this?

I could either do that or fix it. Basically, this assert doesn't make sense and the way we copy vector probably needs a slight change
https://github.com/dotnet/runtime/blob/release/7.0/src/coreclr/jit/gentree.cpp#L21833-L21836

I believe I just need to change it to
image

**Comment by tannergooding** [comment]
> I could either do that or fix it

Just iterating what we chatted about on Teams. We'll want to fix it here, but if you could open an issue for it as well that'd be great, as I need to look at backporting the fix for .NET 7

Thankfully its a small fix (basically 1-line change + tweaking the assert slightly).

**Comment by DeepakRajendrakumaran** [comment]
> > I could either do that or fix it
> 
> Just iterating what we chatted about on Teams. We'll want to fix it here, but if you could open an issue for it as well that'd be great, as I need to look at backporting the fix for .NET 7
> 
> Thankfully its a small fix (basically 1-line change + tweaking the assert slightly).

I'll add a fix here to this PR.

For backporting - https://github.com/dotnet/runtime/issues/85132

**Comment by BruceForstall** [comment]
Do we need new tests to go along with this?

@tannergooding It looks like we're missing tests for lots of the AVX512 intrinsics?

**Comment by tannergooding** [comment]
> @tannergooding It looks like we're missing tests for lots of the AVX512 intrinsics?

This is just doing light-up for `Vector512<T>` which has existing tests covering all exposed APIs under:
* https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/General/Vector512
* https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/General/Vector512_1

-- Noting, they are generated at build time using the tables in https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_General.cs#L2016-L2693

We also have libraries side tests for those APIs here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector512Tests.cs

The platform specific intrinsics have coverage here: https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/X86_Avx512 and are generated from https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.cs#L1094-L1320

I've been ensuring we have test coverage for all the newly introduced things. We do want to add a regression test for the bug fix this PR includes, however. @DeepakRajendrakumaran That should go under https://github.com/dotnet/runtime/tree/main/src/tests/JIT/Regression/JitBlue and follow the general template as the others (name it `Runtime_85132`, add the validation logic in question, and have it return `100` as the exit code for success).

**Comment by BruceForstall** [comment]
> https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_General.cs#L2016-L2693

This table doesn't have an entry for Shuffle. Am I missing something?

**Comment by tannergooding** [comment]
They are mostly on the library side due to complexity in the cross platform implementation: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector512Tests.cs#L2263-L3370

The platform specific APIs are a bit less complex and are table generated:
* https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.cs#L1182-L1185
* https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.cs#L1279-L1284

**Comment by DeepakRajendrakumaran** [comment]
> > @tannergooding It looks like we're missing tests for lots of the AVX512 intrinsics?
> 
> This is just doing light-up for `Vector512<T>` which has existing tests covering all exposed APIs under:
> 
> * https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/General/Vector512
> * https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/General/Vector512_1
> 
> -- Noting, they are generated at build time using the tables in https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_General.cs#L2016-L2693
> 
> We also have libraries side tests for those APIs here: https://github.com/dotnet/runtime/blob/main/src/libraries/System.Runtime.Intrinsics/tests/Vectors/Vector512Tests.cs
> 
> The platform specific intrinsics have coverage here: https://github.com/dotnet/runtime/tree/main/src/tests/JIT/HardwareIntrinsics/X86_Avx512 and are generated from https://github.com/dotnet/runtime/blob/main/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_X86.cs#L1094-L1320
> 
> I've been ensuring we have test coverage for all the newly introduced things. We do want to add a regression test for the bug fix this PR includes, however. @DeepakRajendrakumaran That should go under https://github.com/dotnet/runtime/tree/main/src/tests/JIT/Regression/JitBlue and follow the general template as the others (name it `Runtime_85132`, add the validation logic in question, and have it return `100` as the exit code for success).

Added test.

**Review by BruceForstall — APPROVED** [review]

File: src/coreclr/jit/gentree.cpp [hunk 1]
```diff
@@ -23176,7 +23176,7 @@ GenTree* Compiler::gtNewSimdShuffleNode(
 #if defined(TARGET_XARCH)
     uint8_t  control   = 0;
     bool     crossLane = false;
-    bool     needsZero = varTypeIsSmallInt(simdBaseType);
+    bool     needsZero = varTypeIsSmallInt(simdBaseType) && (simdSize != 64);
```
Comment (tannergooding) [inline src/coreclr/jit/gentree.cpp:23179]: This is correct for the PR, but just noting we'll want to update the other `simdSizes` to use the full-width permutations as well, long term.

**Review by tannergooding — APPROVED** [review]

**Pull request merged**